src/cnaas_nms/confpush/firmware.py

```diff
@@ -268,9 +268,9 @@ def device_upgrade(download: Optional[bool] = False,
         with sqla_session() as session:
             dev: Device = session.query(Device).\
                 filter(Device.hostname == device).one_or_none()
-            if not dev or dev.device_type != DeviceType.ACCESS:
-                raise Exception('Invalid device type: {}'.format(device))
-            if not dev or dev.platform != 'eos':
+            if not dev:
+                raise Exception('Could not find device: {}'.format(device))
+            if dev.platform != 'eos':
                 raise Exception('Invalid device platform: {}'.format(device))
```

Comment:
maybe this should be: 'Invalid device platform: {}'.format(dev.platform) to show the invalid platform? Or even better: 'Invalid device platform "{}" for device: {}'.format(dev.platform, device)

Reply:
Changed it.